src/main/java/org/jmxtrans/embedded/EmbeddedJmxTrans.java

```diff
@@ -248,18 +244,8 @@ public synchronized void stop() throws Exception {
             return;
         }
         try {
-            collectScheduledExecutor.shutdown();
-            try {
-                collectScheduledExecutor.awaitTermination(getQueryIntervalInSeconds(), TimeUnit.SECONDS);
```

Comment:
There is no need to await for the tasks termination, as one last collect and export will be executed anyways. If a collect and/or export is interrupted, not data will be lost, because we will collect again then export again.
```awaitTermination``` is only making the application (and the JVM) longer to stop.